htdocs/core/lib/files.lib.php

```diff
@@ -2293,7 +2295,7 @@ function dol_check_secure_access_document($modulepart, $original_file, $entity,
 
 	// Find the subdirectory name as the reference. For exemple original_file='10/myfile.pdf' -> refname='10'
 	if (empty($refname)) {
-		$refname = basename(dirname($original_file)."/");
+		$refname = basename(dirname($original_file, 2)."/");
```

Comment:
If you add ,2 parameter it means that for original_file = a/b/c, dirname(original_file, 2)  will return 
a
But here we want 
b
Old code seems good.

Reply:
I partially agree with you as I am not experienced as you are in Dolibarr. But here...
a= Project Reference
b= Task Reference inside the above project
c= Filename

Dolibarr creates directory for each and every project to store linked files in it. Now tasks files are stored new directory inside the project directory. So you must know the **_a_**, before you go to **_b_**. I have fixed it and it is completely working in my system pretty good. I will suggest to check once again. Thank you.

Reply:
Good morning,

Dolibarr V 14.0.2 latest update already installed and problem persisting.

I've also changed code following your instructions but still facing the same problem. The most surprising is that depending in system, problem is persisting or not. I mean, I tested 2 desktop PCs, 3 mobile/tablets and all of them working fine, but from laptops (Lenovo & MacBook Air) we are facing problems, no matter from which country do we try or the navigator we use (Chrome or Edge). I don't know if it makes sense, but seems that problems are generated by laptops??!!

Best regards and grateful, hope you could help us. Thanks.